test/unit/InsuranceFundingRate.js

```diff
@@ -1,149 +1,49 @@
+    tracer.setLiquidationContract(accounts[0].address)
 
-    let maxLeverage = ethers.utils.parseEther("12.5")
-    let deleveragingCliff = ethers.utils.parseEther("20") // 20 percent
-    let lowestMaxLeverage = ethers.utils.parseEther("12.5") // Default -> Doesn't go down
-    let insurancePoolSwitchStage = ethers.utils.parseEther("1") // Switches mode at 1%
-    let liquidationGasCost = 63516
+    // connect to insurance as liquidation contract
+    let insurance = await getInsurance(tracer)
+    insurance = await insurance.connect(accounts[0])
 
-    const tracer = await tracerContractFactory.deploy(
-        ethers.utils.formatBytes32String("TEST/USD"),
-        testToken.address,
-        deployer, // Dummy address so it is not address(0)
-        maxLeverage,
-        1,
-        1,
-        deployer, // Dummy address so it is not address(0)
-        deleveragingCliff,
-        lowestMaxLeverage,
-        insurancePoolSwitchStage,
-        liquidationGasCost
-    )
-
-    let mockTracer = await smockit(tracer)
-
-    // mock tracer calls that are needed
-    // get balance for this account to return 0
-    // NOTE: If any test changes mocks, due to Hardhat fixture optimisations,
-    // the mock defaults set here WILL NOT be returned. You need to manually
-    // change the mock state back to its expected value at the end of the test.
-    mockTracer.smocked.getBalance.will.return.with({
-        position: { quote: 0, base: 0 }, //quote, base
-        totalLeveragedValue: 0, //total leverage
-        lastUpdatedIndex: 0, //last updated index
-        lastUpdatedGasPrice: 0, //last updated gas price
-    })
-
-    // token to return the testToken address
-    mockTracer.smocked.tracerQuoteToken.will.return.with(testToken.address)
-
-    // leveraged notional value to return 100
-    mockTracer.smocked.leveragedNotionalValue.will.return.with(
-        ethers.utils.parseEther("100")
-    )
-
-    // quote token decimals
-    mockTracer.smocked.quoteTokenDecimals.will.return.with(18)
-
-    // deposit and withdraw to return nothing
-    mockTracer.smocked.deposit.will.return()
-    mockTracer.smocked.withdraw.will.return()
-
-    // deploy insurance using mock tracer
-    const Insurance = await ethers.getContractFactory("Insurance")
-    let insurance = await Insurance.deploy(mockTracer.address)
-    await insurance.deployed()
     return {
-        testToken,
-        mockTracer,
-        insurance,
+        quoteToken: await getQuoteToken(tracer),
+        tracer: tracer,
+        insurance: insurance,
     }
 })
 
 describe("Unit tests: Insurance.sol", function () {
     let accounts
-    let testToken
-    let mockTracer
+    let quoteToken
+    let tracer
     let insurance
 
     beforeEach(async function () {
-        const _setup = await setup()
-        testToken = _setup.testToken
-        mockTracer = _setup.mockTracer
-        insurance = _setup.insurance
+        ;({ quoteToken, tracer, insurance } = await setupTests())
```

Comment:
This is some funky formatting, is this intentional?

Reply:
So here I'm directly destructuring the object returned by setupTests(). This allows us to skip initialising the `_setup` constant.

JS requires direct destructuring assignments to be wrapped in parentheses. The semicolon is required by the linter as it prevents the line accidently triggering a function in a previous line.

Reply:
Interesting, for some reason I was under the impression you could just use `{ a, b } = someFunc()`. This is really cool though and good info, thanks

Reply:
one of the reasons im bullish on semicolons required in linting rules

Reply:
i believe you only need parentheses around destructuring if you are assigning to variables that are already initialised:

```
const { a, b, c } = somefunc()
```

vs 

```
let a;
let b;
let c;

({ a, b, c }) = somefunc()
```